api/room_handler.go

```diff
@@ -375,14 +375,35 @@ func (h *RoomListByMetricHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		limit,
 		mr,
 	)
+
 	if err != nil {
 		status := http.StatusInternalServerError
 		logger.WithError(err).Error("list by metrics handler failed")
 		h.App.HandleError(w, status, "list by metrics handler error", err)
 		return
 	}
 
-	bytes, err := json.Marshal(map[string]interface{}{"rooms": rooms})
+	// Get list of ready and occupied rooms (excluding creating and terminating rooms from response)
+	readyAndOccupiedRooms, err := models.GetAllRegisteredRoomsByStatus(
+		h.App.RedisClient.Client,
+		params.SchedulerName,
+		[]string{models.StatusReady, models.StatusOccupied})
+
+	if err != nil {
+		logger.WithError(err).Error("failed to list available pods on namespace")
+	}
+
+	// remove terminating and creating rooms
```

Comment:
The problem here is that, since you are filtering out unavailable rooms you'll probably end up returning less rooms that expected (`limit` param). Also, `GetAllRegisteredRoomsByStatus` can be quite expensive if a scheduler has many rooms, specially since this route can be called quite frequently. Could we pass a list of rooms and validate which are ready or occupied? Or maybe filter out creating/terminating ones? We'll probably have less rooms being returned in the second case.

The best approach IMO would be to receive an array of rooms and check using redis `mult` op of they fit the criteria (ready or occupied). If we have less rooms than necessary try it again.

Finally, I'd move this to `models.GetRoomsByMetric` since this won't be useful if we're using this if metric is not cpu or mem.